samples/apps/logging/logging.cpp

```diff
@@ -1023,11 +1033,48 @@ namespace loggingapp
           return;
         }
 
+        const auto indexed_txid =
+          index_per_private_key->get_indexed_watermark();
+        if (indexed_txid.seqno < to_seqno)
+        {
+          index_per_private_key->extend_index_to(
+            {view_of_final_seqno, to_seqno});
+          ctx.rpc_ctx->set_response_status(HTTP_STATUS_ACCEPTED);
+          static constexpr size_t retry_after_seconds = 3;
+          ctx.rpc_ctx->set_response_header(
+            http::headers::RETRY_AFTER, retry_after_seconds);
+          ctx.rpc_ctx->set_response_header(
+            http::headers::CONTENT_TYPE, http::headervalues::contenttype::TEXT);
+          ctx.rpc_ctx->set_response_body(fmt::format(
+            "Still constructing index for private records on key {} - indexed "
+            "to {}/{}",
+            id,
+            indexed_txid.seqno,
+            to_seqno));
```

Comment:
Just double-checking my understanding here: we report the key (`id`) in the error message but the strategy does not actually look at the key and indexes all keys in the target KV map? And if it could only index some keys, `get_indexed_watermark()` would have to take `id` as an argument?

Reply:
Yes, we're just reporting the `id` to be as verbose as possible. The indexing strategy is currently per-table, but we could write a strategy that only cared about a single key. Then the `id` would be provided where we construct the strategy (where we currently only give the table name).